__tests__/app.test.js

```diff
@@ -31,7 +31,53 @@ describe("GET /api/topics", () => {
       .get("/api/topics")
       .expect(200)
       .then(({ body }) => {
-        expect(body.topics).toEqual(data.topicData);
+        expect(body.topics).toHaveLength(3);
+        body.topics.forEach(({ description, slug }) => {
+          expect(typeof description).toBe("string");
+          expect(typeof slug).toBe("string")
+        })
       });
   });
-})
\ No newline at end of file
+})
+
+describe("GET /api/articles/:article_id", () => {
+  test("200: Responds with an object of the article with the correct article id", () => {
+    return request(app)
+      .get("/api/articles/1")
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.article).toHaveLength(1);
+        body.article.forEach(({ article_id, title, topic, author, body, created_at, votes, article_img_url }) => {
+          expect(typeof article_id).toBe("number")
+          expect(typeof title).toBe("string")
+          expect(typeof topic).toBe("string")
+          expect(typeof author).toBe("string")
+          expect(typeof body).toBe("string")
+          expect(typeof created_at).toBe("string")
+          expect(typeof votes).toBe("number")
+          expect(typeof article_img_url).toBe("string")
+        })
+      });
+  });
+  test('GET:404 sends an appropriate status and error meaasge when given a valid bot non-existent id', () => {
```

Comment:
What about when I try to use bananas as an article id? we should have a test for this scenario!